src/attribute.ts

```diff
@@ -176,9 +207,11 @@ export class Attribute {
     }
 
     if (!this.isArray) {
-      throw new Error(`trying to add an item to a non-array attribute`)
+      throw new Error(
+        `trying to add an item to non-array attribute ${this.path}`,
+      )
     }
-    const item = Attribute.from(this)
+    const item = Attribute.from(attr || this)
 
     item.definition.max = '1'
```

Comment:
I don't think so but just do be sure we don't miss anything.
Couldn't we have an extension which is directly an array?

Reply:
no it is not possible to my knowledge. A complex-type/resource always have a child "extension" which is an array of `Extension` resources (you cannot have arrays of arrays imo). You may add an extension to a primitive type by adding an attribute prefixed by an underscore which is an `Element` which also has an array of extensions.